completers/common/xh_completer/cmd/root.go

```diff
@@ -0,0 +1,99 @@
+	rootCmd.Flags().Bool("multipart", false, "Like --form, but force a multipart/form-data request even without files")
+	rootCmd.Flags().Bool("offline", false, "Construct HTTP requests without sending them anywhere")
+	rootCmd.Flags().StringP("output", "o", "", "Save output to FILE instead of stdout")
+	rootCmd.Flags().String("pretty", "", "Controls output processing")
+	rootCmd.Flags().StringP("print", "p", "", "String specifying what the output should contain")
+	rootCmd.Flags().String("proxy", "", "Use a proxy for a protocol. For example: --proxy https:http://proxy.host:8080")
+	rootCmd.Flags().BoolP("quiet", "q", false, "Do not print to stdout or stderr")
+	rootCmd.Flags().String("raw", "", "Pass raw request data without extra processing")
+	rootCmd.Flags().String("resolve", "", "Override DNS resolution for specific domain to a custom IP")
+	rootCmd.Flags().String("response-charset", "", "Override the response encoding for terminal display purposes")
+	rootCmd.Flags().String("response-mime", "", "Override the response mime type for coloring and formatting for the terminal")
+	rootCmd.Flags().String("session", "", "Create, or reuse and update a session")
+	rootCmd.Flags().String("session-read-only", "", "Create or read a session without updating it form the request/response exchange")
+	rootCmd.Flags().String("ssl", "", "Force a particular TLS version")
+	rootCmd.Flags().BoolP("stream", "S", false, "Always stream the response body")
+	rootCmd.Flags().StringP("style", "s", "", "Output coloring style")
+	rootCmd.Flags().Int("timeout", 0, "Connection timeout of the request")
+	rootCmd.Flags().BoolP("verbose", "v", false, "Print the whole request as well as the response")
+	rootCmd.Flags().String("verify", "", "If 'no', skip SSL verification. If a file path, use it as a CA bundle")
+	rootCmd.Flags().BoolP("version", "V", false, "Print version")
+
+	rootCmd.MarkFlagsMutuallyExclusive("json", "form", "multipart")
+
+	carapace.Gen(rootCmd).FlagCompletion(carapace.ActionMap{
+		"auth-type":         carapace.ActionValues("basic", "bearer", "digest"),
+		"history-print":     carapace.ActionValues("h", "b", "m").UniqueList(""),
+		"http-version":      carapace.ActionValues("1.0", "1.1", "2", "2-prior-knowledge"),
+		"output":            carapace.ActionFiles(),
+		"pretty":            carapace.ActionValues("all", "colors", "format", "none"),
+		"print":             carapace.ActionValues("h", "b", "m").UniqueList(""),
+		"session":           carapace.ActionFiles(),
+		"session-read-only": carapace.ActionFiles(),
+		"ssl":               carapace.ActionValues("auto", "tls1", "tls1.1", "tls1.2", "tls1.3"),
+		"style":             carapace.ActionValues("auto", "solarized", "monokai", "fruity"),
+		"verify": carapace.Batch(
+			carapace.ActionValuesDescribed("no", "Skip SSL verification"),
+			carapace.ActionFiles(),
+		).ToA(),
+	})
+
+	carapace.Gen(rootCmd).PositionalCompletion(
+		http.ActionRequestMethods(),
```

Comment:
Request methods are optional and tthen here's the whole request items stuff.
I've done that for [httpie](https://github.com/carapace-sh/carapace-bin/blob/master/completers/common/http_completer/cmd/root.go) and this should be similar, but it gets quite nasty quickly.

This is fine for now though.